packages/core/src/dialects/sqlite/query-interface.ts

```diff
@@ -293,4 +382,162 @@ export class SqliteQueryInterfaceTypeScript extends AbstractQueryInterface {
+   * It will create a backup of the table, drop the table afterwards and create a
+   * new table with the same name but without the obsolete column.
+   *
+   * @param tableName
+   * @param attributeName
+   * @param options
+   */
+  async removeColumn(tableName: TableNameOrModel, attributeName: string, options?: RemoveColumnOptions) {
+    const fields = await this.describeTable(tableName, { ...options });
+    delete fields[attributeName];
+
+    return this.alterTableInternal(tableName, fields, { ...options });
+  }
+
+  /**
+   * A wrapper that fixes SQLite's inability to change columns from existing tables.
+   * It will create a backup of the table, drop the table afterwards and create a
+   * new table with the same name but with a modified version of the respective column.
+   *
+   * @param tableName
+   * @param columnName
+   * @param dataTypeOrOptions
+   * @param options
+   */
+  async changeColumn(
+    tableName: TableNameOrModel,
+    columnName: string,
+    dataTypeOrOptions: DataType | AttributeOptions,
+    options?: QueryRawOptions,
+  ) {
+    const columns = await this.describeTable(tableName, { ...options });
+    for (const column of Object.values(columns)) {
+      // This is handled by copying indexes over,
+      // we don't use "unique" because it creates an index with a name
+      // we can't control
+      delete column.unique;
+    }
+
+    Object.assign(columns[columnName], this.sequelize.normalizeAttribute(dataTypeOrOptions));
+
+    return this.alterTableInternal(tableName, columns, { ...options });
+  }
+
+  /**
+   * A wrapper that fixes SQLite's inability to rename columns from existing tables.
+   * It will create a backup of the table, drop the table afterwards and create a
+   * new table with the same name but with a renamed version of the respective column.
+   *
+   * @param tableName
+   * @param attrNameBefore
+   * @param attrNameAfter
+   * @param options
+   */
+  async renameColumn(tableName: TableNameOrModel, attrNameBefore: string, attrNameAfter: string, options?: QueryRawOptions) {
+    const fields = await this.assertTableHasColumn(tableName, attrNameBefore, options);
+
+    fields[attrNameAfter] = { ...fields[attrNameBefore] };
+    delete fields[attrNameBefore];
+
+    const sql = this.queryGenerator._replaceColumnQuery(tableName, attrNameBefore, attrNameAfter, fields);
+    await this.#internalQueryInterface.executeQueriesSequentially(sql, { ...options, raw: true });
+  }
+
+  /**
+   * Alters a table in sqlite.
+   * Workaround for sqlite's limited alter table support.
+   *
+   * @param tableName
+   * @param columns
+   * @param options
+   */
+  private async alterTableInternal(
+    tableName: TableNameOrModel,
+    columns: SQLiteColumnsDescription,
+    options?: QueryRawOptions,
+  ) {
+    const table = this.queryGenerator.extractTableDetails(tableName);
+
+    return withSqliteForeignKeysOff(this.sequelize, options ?? {}, async () => {
```

Comment:
prefer EMPTY_OBJECT over `{}`

Reply:
I updated `withSqliteForeignKeysOff` to accepted `QueryRawOptions` or `undefined` so the `{}` isn't required.